network/manager.go

```diff
@@ -192,13 +193,25 @@ func (nm *networkManager) restore(isRehydrationRequired bool) error {
 	// Read any persisted state.
 	err := nm.store.Read(storeKey, nm)
 	if err != nil {
+		var syntaxErr *json.SyntaxError
 		if err == store.ErrKeyNotFound {
 			logger.Info("network store key not found")
 			// Considered successful.
 			return nil
 		} else if err == store.ErrStoreEmpty {
 			logger.Info("network store empty")
 			return nil
+		} else if errors.As(err, &syntaxErr) {
+			// if null chars detected or failed to parse, state is unrecoverable; delete it
+			logger.Error("Failed to parse corrupted state, deleting", zap.Error(err))
+			contents, readErr := nm.store.Dump()
+			if readErr != nil {
+				logger.Error("Could not read corrupted state", zap.Error(readErr))
+			} else {
+				logger.Info("Logging state", zap.String("stateFile", contents))
```

Comment:
nit: logging corrupted state.

Reply:
We log the corrupted state for debugging as per the suggestion by Michael above. Is there some drawback of doing this?